From 59fb4c93d3afabde8b0573045082c0a3bd3023d9 Mon Sep 17 00:00:00 2001
From: Stephen Hutchinson <qyot27@gmail.com>
Date: Wed, 19 Jul 2023 14:19:37 -0400
Subject: [PATCH] Version: Fix RELEASE_TARBALL case in mixed chroot environment

Uncovered by haikuports, because that environment uses a git
directory during the build control process, but the chroot it drops
into for the build does not have git available.  This resulted in
the RELEASE_TARBALL check failing because the `.git` directory
had reappeared, but then git wasn't there so Version.cmake failed.
---
 avs_core/CMakeLists.txt | 2 +-
 avs_core/Version.cmake  | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/avs_core/CMakeLists.txt b/avs_core/CMakeLists.txt
index eef0f8b0..ea100af8 100644
--- a/avs_core/CMakeLists.txt
+++ b/avs_core/CMakeLists.txt
@@ -202,7 +202,7 @@ ADD_CUSTOM_TARGET(
 )
 ADD_DEPENDENCIES("AvsCore" VersionGen)

-if (NOT EXISTS "${CMAKE_SOURCE_DIR}/.git")
+if (NOT EXISTS "${CMAKE_SOURCE_DIR}/.git" OR NOT GIT_FOUND)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DRELEASE_TARBALL")
 endif()

diff --git a/avs_core/Version.cmake b/avs_core/Version.cmake
index 33c51e7c..6cd792ab 100644
--- a/avs_core/Version.cmake
+++ b/avs_core/Version.cmake
@@ -1,4 +1,5 @@
-if (EXISTS ${REPO}/.git)
+FIND_PACKAGE(Git)
+if (EXISTS "${REPO}/.git" AND GIT_FOUND)
 EXECUTE_PROCESS(
     COMMAND "${GIT}" --git-dir=${REPO}/.git  rev-list --count HEAD
     OUTPUT_VARIABLE AVS_SEQREV
--
2.37.3
